### ISSUE TITLE
Virtual fails when including cjs virtual file when targeting esm output format

### ISSUE THREAD

**Issue by CxRes** [rollup/plugins#651]
- Rollup Plugin Name:  Virtual
- Rollup Plugin Version: 2.0.2
- Rollup Version: 2.33.1
- Operating System (or Browser): Windows 10 x64
- Node Version: 14.14.0
- Link to reproduction: [https://repl.it/@CxRes/rollup-plugin-virtual-bug#output/real.js](https://repl.it/@CxRes/rollup-plugin-virtual-bug)

I am trying to do something a bit funky - to allow esm code to run in nwjs which does not support import of node builtin modules :(. When doing so, **Including the cjs module with the Virtual plugin fails. But this works perfectly in case I am including a real file with identical content.**

In the example, I have a module that imports `path` (in files `input-*.js`). Because, nwjs does not allow me to import path, I "require" it in another file and ask Rollup to bundle that instead. With a real file  (`./local-path.js`) this works as expected. But when using the virtual plugin with identical content, Rollup complains that there there is no default export (which is technically true since one uses `module.exports`, but then Rollup should do its magic to fix this just like the regular file case). 

### Expected Behavior

At the very least, both cases should behave consistently. Including a module using the virtual plugin should bundle identically as including another file. 

In the worst case scenario, both should fail!

### Actual Behavior

Including the cjs module using the Virtual plugin throws complaining about a missing default export. 

NB: Including the cjs module as a normal file is handled well by Rollup resulting in the file `output/real.js` (It is my intention to preserve the require statement in this file even when I am bundling as esm as explained above. If other import/export statements existed I want them to remain!)

**Comment by shellscape** [comment]
Thanks for the issue. We're a little short on help with the virtual plugin, so if you're interested in opening a PR we'd welcome it!

**Comment by CxRes** [comment]
The use of stale bot is so annoying. It demonstrates more that devs don't actually care for the users. Clearly, this issue needs to remain open as the reason for not acting on it is absence of a maintainer (which I can understand). Closing the issue is feels like someone slamming the door on your face!

**Comment by shellscape** [comment]
@CxRes it's best not to take repository maintenance so personal. The issue remains searchable and for now, open to all. I would ask that you refrain from emotionally charged replies in the future and only comment if you have something constructive to add to the issue for triage, development, or reproduction discussion. If you cannot honor that request, we'll have no choice but to lock the issue. This issue remains open to Pull Requests from the community, but as it has sat without attention for 60 days, we'll keep it closed to prevent the list of active issues from growing beyond a reasonable number.

**Comment by CxRes** [comment]
@shellscape Am I not free to voice my views now! I have not used language that is anyway offensive and expressed merely a strongly held opinion that is perfectly pertinent and constructive (perhaps not in your narrow technical definition). Your language that " I would ask that you refrain..." and that you are going to the lengths of marking it as off-topic (to hide the comment unless someone really looks) and further the threat to lock is all representative of your attitudes towards your contributors and community. Basically your commentary is akin to a pathological corporation and not a friendly open source project. Which is a shame given how much I have enjoyed the work of @Rich-Harris.
 
I took the time to file a bug and create a use case for it, it is a contribution too, though a very small one. People who may come across this bug should be able to find it easily. Most people don't wade through closed issues and will most likely waste more of their time as a result. Closing the issue says that my contribution was unwelcome, maintainer don't care about tracking problems in code and time of other who may encounter that same bug is not important (and not that the project can use more help).

I was actually going to contribute in a day or two by fixing up a small unrelated Rollup plugin that had not been updated in five years. But why bother when I shall be met with such disdain and apathy. Don't worry, this is the last time I will ever bother with this repository, so your locking shall be inconsequential (and will prevent someone else who encounters the same issue from commenting).  Good bye!
